### PR TITLE
fix: handled fetcher function gracefully

### PR DESCRIPTION
**Pull request by shamim-io**
closes https://github.com/hirosystems/explorer/issues/1154

Handled fetcher function gracefully

Previous code: 
Screenshot 2023-05-30 at 11 24 55 PM

Modified code: 
Screenshot 2023-05-30 at 11 35 51 PM

Files changed
`src/common/api/fetch.ts`

### CODE REVIEW

**Review by He1DAr — APPROVED** [review]

File: src/common/api/fetch.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-    headers: { ...initHeaders, ...HIRO_HEADERS },
-  });
+  try {
+    return await fetch(input, {
+      credentials: 'omit',
+      ...init,
+      headers: { ...initHeaders, ...HIRO_HEADERS },
+    });
+  } catch (error) {
+    throw new Error(`Network error: ${error.message}`);
```
Comment (He1DAr) [inline src/common/api/fetch.ts:12]: Hi @shamim-io! Thank you for submitting this PR. Could you explain the motivation behind this change? It looks like the error is being rethrown.

**Review by He1DAr — CHANGES_REQUESTED** [review]

File: src/common/api/fetch.ts (hunk 1 above)
Comment (shamim-io) [reply src/common/api/fetch.ts:12]: @He1DAr If you think that, its not a required change. Feel free to close it :)

**Comment by He1DAr** [comment]
Thank you for submitting this PR!
I think the new implementation will behave exactly the same, except it adds "Network error" to the error message when it throws. Here's an example:
image
image
Since we want to keep throwing the error, I suggest we close the PR. Please let me know your thoughts on this.

**Comment by shamim-io** [comment]
I agree with you. Thanks @He1DAr for the clarification. Closing the PR!

**Pull request closed without merging**